### DISCUSSION TITLE
why doafter in shared not firing event for client

### DISCUSSION THREAD

**Discussion by deltanedas** [space-wizards/space-station-14#14967] [Questions] [answered]
i have a doafter that is created in shared code and printf debugged that is indeed called
my doafter event handler is only called on the server also printf debugged
why is this happening and how can i fix it

**Comment by ElectroJr** [comment, marked as answer]
Even though DoAfters were moved to shared in the refactor, they still didn't fully support networking & prediction. #14609 should fix that, you should probably just wait for that to be finished.

**Comment by deltanedas** [comment]
oh nice
